### PR TITLE
Correctly add BspFetchTestEnvironmentTask to run configurations

### PR DESCRIPTION
**Pull request by tpasternak**
The Scala plugin tries to install BspFetchTestEnvironmentTask on
`runConfigurationAdded` event. Unfortunately, it requires
RunConfigurationEx service to be installed, and the event happens
during the initialization of it.

Apart from that, the `beforeRunTasks.add` was previously called
uncoditionally. This caused multiple additions of the task.

### CODE REVIEW

**Comment by unkarjedy** [comment]
@tpasternak Please update from the master. I've pushed the fix for cyclic services declaration yesterday

**Comment by tpasternak** [comment]
Thanks, rebased and adapted to your solution (which is probably more safe than my previous one)

File: bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchEnvironmentTaskInstaller.scala [hunk 1]
```diff
@@ -1,38 +1,40 @@
...
+        beforeRunTasks.add(task)
       }
     }
   }
 
   override def stateLoaded(runManager: RunManager, isFirstLoadState: Boolean): Unit = {
     if (isFirstLoadState) {
-      this.runManagerOpt = Some(runManager.asInstanceOf[RunManagerEx])
-      settingsToInit.foreach(runConfigurationAdded)
+      runManagerEx.foreach {
```
Comment (unkarjedy) [inline bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchEnvironmentTaskInstaller.scala:30]: IMO we should use the instance provided in parameters to be clear

Comment (unkarjedy) [reply bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchEnvironmentTaskInstaller.scala:30]: so looks like no need in `project` parameter in the Service

File: bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchEnvironmentTaskInstaller.scala [hunk 2]
```diff
@@ -1,38 +1,40 @@
...
+        runManager => settingsToInit.foreach(installFetchEnvironmentTask(_, runManager))
```
Comment (unkarjedy) [inline bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchEnvironmentTaskInstaller.scala:31]: `settingsToInit` are always empty now
Shouldn't we patch the loaded configurations?
For some reason when initializing `RunManager` platform calls `runConfigurationAdded` for the saved run configurations (not after the service initialization). In my fix commit, I postponed the initialization of saved run configurations until `stateLoaded` method.

**Review by jastice — APPROVED** [review]

**Comment by unkarjedy** [comment]
I forgot to press "Add a review" yesterday, nevermind, we will fix ^

File: bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchEnvironmentTaskInstaller.scala (hunk 2 above)
Comment (tpasternak) [reply bsp/src/org/jetbrains/bsp/project/test/environment/BspFetchEnvironmentTaskInstaller.scala:31]: Yes, sorry for that. It was clear for me, that the `settingsToInit` should be filled in `runConfigurationAdded` handler, but I broke it while refactoring code.

**Comment by tpasternak** [comment]
@unkarjedy I can fix it right now, but please let me know, I wouldn't like to collide with you again

**Comment by jastice** [comment]
@tpasternak I just pushed a fix

**Comment by tpasternak** [comment]
Ok, thanks and sorry

**Pull request merged**